cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -566,57 +635,102 @@ def clean_docket_number(docket_number: str) -> str:
     return docket_number
 
 
-def clean_body_content(case_body: str) -> str:
+def clean_body_content(case_body: str, harvard: bool = False) -> str:
     """Strip all non-alphanumeric characters
 
     :param case_body: Opinion text
+    :param harvard: Are we harvard xml data
     :return:Opinion text with only alphanumeric characters
     """
     soup = BeautifulSoup(case_body, "lxml")
-    return re.sub(r"[^a-zA-Z0-9]", "", soup.text)
+    if not harvard:
+        opinion_text = soup.text
+    else:
+        opinions = []
+        for op in soup.find_all("opinion"):
+            opinions.append(op.text)
+        opinion_text = "".join([op.text for op in soup.find_all("opinion")])
+
+    return re.sub(r"[^a-zA-Z0-9]", "", opinion_text.lower())
 
 
 def match_based_text(
-    case_body: str,
+    harvard_characters: str,
     docket_number: str,
     case_name_full: str,
     possible_cases: List,
+    case_name_abbreviation: str,
+    citation: Citation,
 ) -> Optional[OpinionCluster]:
     """Compare CL text to Harvard content to establish duplicates
 
-    :param case_body: Harvard case body to compare
+    :param harvard_characters: Harvard stripped characters to compare
     :param possible_cases: List of opinions to check against
     :param docket_number: The docket number
     :param case_name_full: The full case name
     :return: OpinionCluster or None
     """
-    harvard_characters = clean_body_content(case_body)
     for case in possible_cases:
         cl_case_body = get_opinion_content(case)
         cl_characters = clean_body_content(cl_case_body)
+        if len(cl_characters) == 0:
+            logger.info(f"Empty Courtlistener opinion cluster: {case.id}")
```

Comment:
A `logger.warn` would be good here too, I think.

Reply:
√